cmd/push.go

```diff
@@ -213,24 +332,109 @@ func (p *PushCmdParams) Run(ctx ActionCtx) (store.Status, error) {
 	if err != nil {
 		return nil, err
 	}
-
 	r := store.NewDetailedReport(true)
-	for _, v := range p.targeted {
-		sub := store.NewReport(store.OK, "push %s to account server", v)
-		sub.Opt = store.DetailsOnErrorOrWarning
-		r.Add(sub)
-		ps, err := p.pushAccount(v, ctx)
-		if ps != nil {
-			sub.Add(store.HoistChildren(ps)...)
+	if p.sysAccUser == "" {
+		for _, v := range p.targeted {
+			sub := store.NewReport(store.OK, "push %s to account server", v)
+			sub.Opt = store.DetailsOnErrorOrWarning
+			r.Add(sub)
+			ps, err := p.pushAccount(v, ctx)
+			if ps != nil {
+				sub.Add(store.HoistChildren(ps)...)
+			}
+			if err != nil {
+				sub.AddError("failed to push account %q: %v", v, err)
+			}
+			if sub.OK() {
+				sub.Label = fmt.Sprintf("pushed %q to account server", v)
+			}
 		}
+	} else {
+		sysAcc, sysAccUser, opt, err := systemAccountUser(ctx, p.sysAccUser)
 		if err != nil {
-			sub.AddError("failed to push account %q: %v", v, err)
+			r.AddError("error obtaining system account user: %v", err)
+			return r, nil
 		}
-		if sub.OK() {
-			sub.Label = fmt.Sprintf("pushed %q to account server", v)
+		nc, err := nats.Connect(p.ASU, opt, nats.Name("nsc-client"))
+		if err != nil {
+			r.AddError("failed to connect: %v", err)
+			return r, nil
+		}
+		defer nc.Close()
+		if len(p.targeted) != 0 {
+			sub := store.NewReport(store.OK, `push to nats-server "%s" using system account "%s" user "%s"`,
+				p.ASU, sysAcc, sysAccUser)
+			r.Add(sub)
+			for _, v := range p.targeted {
+				subAcc := store.NewReport(store.OK, "push %s to nats-server with nats account resolver", v)
+				sub.Add(subAcc)
+				if raw, err := ctx.StoreCtx().Store.Read(store.Accounts, v, store.JwtName(v)); err != nil {
+					subAcc.AddError("failed to read account %q: %v", v, err)
+				} else {
+					resp := multiRequest(nc, subAcc, "push account", "$SYS.REQ.CLAIMS.UPDATE", raw,
```

Comment:
I see these subs from the system account:

```
[8752] 2020/10/20 03:12:24.847901 [TRC] SYSTEM - <<- [SUB $SYS.ACCOUNT.*.CLAIMS.UPDATE  32]
[8752] 2020/10/20 03:12:24.847920 [TRC] SYSTEM - <<- [SUB $SYS.REQ.ACCOUNT.*.CLAIMS.UPDATE  33]
[8752] 2020/10/20 03:12:24.847925 [TRC] SYSTEM - <<- [SUB $SYS.REQ.ACCOUNT.*.CLAIMS.LOOKUP  34]
[8752] 2020/10/20 03:12:24.847934 [TRC] SYSTEM - <<- [SUB $SYS.REQ.CLAIMS.PACK responder 35]
```

But do not see `$SYS.REQ.CLAIMS.UPDATE`, is this on a different branch from the nats-server?

Reply:
This depends on nats-server changes, currently under review
I want to land the update to this PR later today.

Reply:
ok thanks, let me know if I can help

Reply:
fixed now